craft_application/services/linter.py

```diff
@@ -0,0 +1,268 @@
+                continue
+
+            try:
+                linter_name, remainder = rule.split(":", 1)
+            except ValueError:
+                continue
+
+            if "=" in remainder:
+                issue_id, glob = remainder.split("=", 1)
+                spec = result.setdefault(
+                    linter_name, IgnoreSpec(ids=set(), by_filename={})
+                )
+
+                spec.by_filename.setdefault(issue_id, set()).add(glob)
+            else:
+                issue_id = remainder
+                spec = result.setdefault(
+                    linter_name, IgnoreSpec(ids=set(), by_filename={})
+                )
+                if isinstance(spec.ids, set):
+                    spec.ids.add(issue_id)
+                else:
+                    pass
+        return result
+
+    @staticmethod
+    def _merge_into(base_cfg: IgnoreConfig, overlay: IgnoreConfig) -> None:
+        """Merge overlay rules into base config, with overlay taking precedence."""
+        for linter_name, over_spec in overlay.items():
+            if linter_name not in base_cfg:
+                base_cfg[linter_name] = IgnoreSpec(
+                    ids=set() if over_spec.ids != "*" else "*",
+                    by_filename={k: set(v) for k, v in over_spec.by_filename.items()},
+                )
+                if over_spec.ids == "*":
+                    continue
+                if isinstance(over_spec.ids, set):
+                    base_cfg[linter_name].ids = set(over_spec.ids)
+                continue
+            base_spec = base_cfg[linter_name]
+
+            if over_spec.ids == "*":
+                base_spec.ids = "*"
+                base_spec.by_filename.clear()
+            elif isinstance(over_spec.ids, set):
+                if base_spec.ids != "*":
+                    if not isinstance(base_spec.ids, set):
+                        base_spec.ids = set()
+                    base_spec.ids.update(over_spec.ids)
+            for issue_id, globs in over_spec.by_filename.items():
+                base_spec.by_filename.setdefault(issue_id, set()).update(globs)
+
+    def pre_filter_linters(
+        self,
+        stage: Stage,
+        ctx: LintContext,  # noqa: ARG002 (reserved for future use)
+        candidates: list[type[AbstractLinter]] | None = None,
+    ) -> list[type[AbstractLinter]]:
+        """App-specific selection hook."""
+        registry = type(self)._class_registry  # noqa: SLF001
+        return list(candidates or registry.get(stage, []))
+
+    def post_filter_issues(
+        self,
+        linter: AbstractLinter,  # noqa: ARG002 (reserved for future use)
+        issues: Iterable[LinterIssue],
+        ctx: LintContext,  # noqa: ARG002 (reserved for future use)
+    ) -> Iterator[LinterIssue]:
+        """App-specific filtering hook."""
+        yield from issues
+
+    def run(
+        self,
+        stage: Stage,
+        ctx: LintContext,
+    ) -> Iterator[LinterIssue]:
```

Comment:
We should have an integration or end-to-end test there that shows a linter erroring out and then an update to the project having it ignored.